pytential/symbolic/pde/scalar.py

```diff
@@ -162,86 +227,104 @@ def operator(self, u):
             # See Hackbusch, https://books.google.com/books?id=Ssnf7SZB0ZMC
             # Theorem 8.2.18b
 
-            amb_dim = self.kernel.dim
             ones_contribution = (
-                    sym.Ones() * sym.mean(amb_dim, amb_dim-1, inv_sqrt_w_u))
+                    sym.Ones() * sym.mean(self.dim, self.dim - 1, inv_sqrt_w_u))
         else:
             ones_contribution = 0
 
-        return (-self.loc_sign*0.5*u
-                + sqrt_w*(
-                    self.alpha*sym.S(self.kernel, inv_sqrt_w_u,
-                        qbx_forced_limit=+1, kernel_arguments=self.kernel_arguments)
-                    - sym.D(self.kernel, inv_sqrt_w_u,
-                        qbx_forced_limit="avg",
-                        kernel_arguments=self.kernel_arguments)
-                    + ones_contribution))
+        def S(density):  # noqa
+            return sym.S(self.kernel, density,
+                    kernel_arguments=self.kernel_arguments,
+                    qbx_forced_limit=+1, **kwargs)
+
+        def D(density):  # noqa
+            return sym.D(self.kernel, density,
+                    kernel_arguments=self.kernel_arguments,
+                    qbx_forced_limit="avg", **kwargs)
+
+        return (
+                -0.5 * self.loc_sign * u + sqrt_w * (
+                    self.alpha * S(inv_sqrt_w_u)
+                    - D(inv_sqrt_w_u)
+                    + ones_contribution
+                    )
+                )
 
 # }}}
 
 
 # {{{ neumann
 
 class NeumannOperator(L2WeightedPDEOperator):
-    """IE operator and field representation for solving Dirichlet boundary
-    value problems with scalar kernels (e.g.
-    :class:`sumpy.kernel.LaplaceKernel`,
-    :class:`sumpy.kernel.HelmholtzKernel`, :class:`sumpy.kernel.YukawaKernel`)
+    """IE operator and field representation for solving Neumann boundary
+    value problems with scalar kernels (e.g. :class:`~sumpy.kernel.LaplaceKernel`,
+    :class:`~sumpy.kernel.HelmholtzKernel`, :class:`~sumpy.kernel.YukawaKernel`)
 
     .. note ::
 
         When testing this as a potential matcher, note that it can only
         access potentials that come from charge distributions having *no* net
-        charge. (This is true at least in 2D.)
+        charge. (This is true at least in 2D)
+
+    Inherits from :class:`L2WeightedPDEOperator`.
 
     .. automethod:: is_unique_only_up_to_constant
     .. automethod:: representation
     .. automethod:: operator
+    .. automethod:: __init__
     """
 
-    def __init__(self, kernel, loc_sign, alpha=None,
-            use_improved_operator=True,
-            laplace_kernel=0, use_l2_weighting=False,
```

Comment:
I removed this `laplace_kernel=0` since it wasn't used and wasn't cleanr what it was for. Any ideas?

Reply:
:shrug: Backward compatibility with some braindeadness? I'm sure that can go.